library/core/src/cmp.rs

```diff
@@ -1032,7 +1047,7 @@ pub macro Ord($item:item) {
 )]
 #[const_trait]
 #[rustc_diagnostic_item = "PartialOrd"]
-pub trait PartialOrd<Rhs: ?Sized = Self>: PartialEq<Rhs> {
+pub trait PartialOrd<Rhs: ?Sized = Self>: ~const PartialEq<Rhs> {
```

Comment:
This change causes the diagnostic regression, but I don't understand why, maybe caused by something in the compiler?
@oli-obk Do you know what causes this / how to fix it?